### ISSUE TITLE
ENH: support for non-nanosecond precision in pd.Timedelta

### ISSUE THREAD

**Issue by veenstrajelmer** [pandas-dev/pandas#59389]
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

When converting matlab datenums, I would expect this should be easily possible with `pd.Timedelta()`, however, several approaches give different errors:
```python
import pandas as pd
# raises OutOfBoundsTimedelta: seconds=62703763200000000000, milliseconds=0, microseconds=0, nanoseconds=0
pd.Timedelta(days=725738)
# raises OutOfBoundsTimedelta: Cannot cast 725738 from D to 'ns' without overflow.
pd.to_timedelta(725738, unit='days')
```

I expect this is because of the nanosecond accuracy. Matlab dates are in days (dtype float) and I would be happy with microsecond (or less) precision. `pd.date_range()` and `pd.Timestamp()` now support non-nanosecond accuracy. Would this also be possible with `pd.Timedelta()`?

### Feature Description

Add support for non-nanosecond accuracy for `pd.Timedelta()`

### Alternative Solutions

It is possible to convert matlab datenums to pandas datetimes with the following approach from https://stackoverflow.com/a/49135037/5578371:
```python
import pandas as pd
import numpy as np
time_wl = pd.to_datetime(np.array([725738])-719529, unit='D').round('s')
```
Gives: `DatetimeIndex(['1987-01-01'], dtype='datetime64[ns]', freq=None)`

However, this loses accuracy and rounding to seconds is required for non-round values.

**Comment by rhshadrach** [comment]
Thanks for the request. For `pd.to_timedelta`, I believe this is a duplicate of #49060.

@jbrockmendel - should `pd.Timedelta` also be added to #49060?

**Comment by veenstrajelmer** [comment]
Thanks for the reply. I actually found that issue in my search, but since it was so old I wrongly assumed it was pre 2.0. I will read better in the future. Either way, feel free to close this issue if needed.

**Comment by jbrockmendel** [comment]
I dont think we should add the keyword to pd.Timedelta, no.  The particular construction from the OP is a tough case; something like `pd.Timedelta(np.timedelta64(N, "D"))` should work as expected

**Comment by jorisvandenbossche** [comment]
In general we already do support non-nanosecond precision for Timedelta, and with the recent changes to change the default resolution for Timedelta (i.e. no longer default to nanoseconds), the example from the OP for the `pd.Timedelta()` constructor also works now:

```python
>>> pd.to_timedelta(725738, unit='days')
Timedelta('725738 days 00:00:00')
>>> pd.Timedelta(days=725738)
Timedelta('725738 days 00:00:00')
```
